File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py

def solution(array, target):
    result = []
    def recur(array, idx, sum_v):
        if idx == len(array):
            result.append(sum_v)
            return
        recur(array, idx + 1, sum_v + array[idx])
        recur(array, idx + 1, sum_v - array[idx])

    recur(array,0,0)
    cnt = result.count(target)
    return cnt

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;
import java.io.*;
import java.math.*;

class Solution {
    static Queue<int[]> que = new ArrayDeque<>();
    public int solution(int[] numbers, int target) {
        int answer = 0;
        
        que.offer(new int[]{0,0});
        while (!que.isEmpty()) {
            int[] temp = que.poll();
            int sum = temp[0];
            int idx = temp[1];

            if (idx == numbers.length){
                if(sum == target){
                    answer++;
                }
                continue;
            }
            que.offer(new int[] { sum + numbers[idx], idx + 1 });
            que.offer(new int[] { sum - numbers[idx], idx + 1 });

            }

            return answer; 
        }
}